.github/workflows/ci.yml

```
@@ -2,41 +2,118 @@ name: CI
+      run: |
+        dnf update -y
+        dnf install -y epel-release
+        dnf config-manager --set-enabled powertools
+        dnf install -y bzip2 bzip2-devel chrpath clang cmake freetype-devel gcc-c++ \
+            git gtk3-devel libstdc++-static libuuid-devel libvorbis-devel \
+            libxml2-devel meson ninja-build ocl-icd-devel openal-soft-devel \
+            opencl-headers openssl-devel patch perl-IPC-Cmd svn unix2dos xz zlib-static
+
+    - uses: actions/checkout@v4
       with:
         submodules: true
 
-    - name: Install dependencies
+    - name: Clone SDL2
+      run: git clone --depth 1 --branch release-$SDL2_VERSION https://github.com/libsdl-org/SDL.git
+
+    - name: Build SDL2 (Release)
       run: |
-        sudo apt-add-repository ppa:hrzhu/sdl2-backport
-        sudo apt-get update
-        sudo apt-get install ninja-build libsdl2-dev
+        cd SDL
+        cmake -B release -G Ninja . -DCMAKE_BUILD_TYPE=Release
+        ninja -C release
+        cd ..
 
     - name: CMake configure (Debug)
-      run: cmake -B debug -G Ninja . -DCMAKE_BUILD_TYPE=Debug
+      run: cmake -B debug -G Ninja . -DCMAKE_BUILD_TYPE=Debug -DSDL2_INCLUDE_DIRS=${GITHUB_WORKSPACE}/SDL/include -DSDL2_LIBRARIES=${GITHUB_WORKSPACE}/SDL/release/libSDL2-2.0.so
 
     - name: Build (Debug)
       run: ninja -C debug
 
     - name: CMake configure (Release)
-      run: cmake -B release -G Ninja . -DCMAKE_BUILD_TYPE=Release
+      run: cmake -B release -G Ninja . -DCMAKE_BUILD_TYPE=Release -DSDL2_INCLUDE_DIRS=${GITHUB_WORKSPACE}/SDL/include -DSDL2_LIBRARIES=${GITHUB_WORKSPACE}/SDL/release/libSDL2-2.0.so
 
     - name: Build (Release)
       run: ninja -C release
 
+    - name: Archive build result
+      uses: actions/upload-artifact@v4
+      with:
+        name: FAudio-SDL2-lib64
+        path: release/libFAudio.so.0
+
+  linux-mingw:
+    name: Rocky Linux MinGW
+    runs-on: ubuntu-latest
+    container:
+      image: rockylinux/rockylinux:8
+    steps:
+      - name: Are we really on Rocky Linux?
+        run: cat /etc/os-release
+
+      - name: Install dependencies
+        run: |
+          dnf update -y
+          dnf install -y epel-release
+          dnf --enablerepo=powertools install -y git xz cmake ninja-build mingw32-gcc-c++ mingw64-gcc-c++
+
+      - uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Download SDL
+        run: |
+          curl -LO https://github.com/libsdl-org/SDL/releases/download/release-$SDL2_VERSION/SDL2-devel-$SDL2_VERSION-mingw.tar.gz
+          tar -xvf SDL2-devel-$SDL2_VERSION-mingw.tar.gz -C ${GITHUB_WORKSPACE}
+
+      - name: CMake configure 32-bit (Release)
+        run: cmake -B release-x86 -G Ninja . -DCMAKE_BUILD_TYPE=Release -DCMAKE_TOOLCHAIN_FILE=.github/mingw-w64-i686.cmake -DSDL2_LIBRARIES=${GITHUB_WORKSPACE}/SDL2-$SDL2_VERSION/i686-w64-mingw32/bin/SDL2.dll -DSDL2_INCLUDE_DIRS=${GITHUB_WORKSPACE}/SDL2-$SDL2_VERSION/i686-w64-mingw32/include/SDL2
```

I _think_ this still works...

```suggestion
        run: mingw32-cmake -B release-x86 -G Ninja . -DCMAKE_BUILD_TYPE=Release  -DSDL2_LIBRARIES=${GITHUB_WORKSPACE}/SDL2-$SDL2_VERSION/i686-w64-mingw32/bin/SDL2.dll -DSDL2_INCLUDE_DIRS=${GITHUB_WORKSPACE}/SDL2-$SDL2_VERSION/i686-w64-mingw32/include/SDL2
```